src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -410,6 +413,9 @@ public DataAccessRequest updateByReferenceIdVersion2(User user, DataAccessReques
             dar.getSubmissionDate(),
             now,
             dar.getData());
+        if (Objects.nonNull(dar.getCollectionId())) {
+            darCollectionDAO.updateDarCollection(dar.getCollectionId(), user.getDacUserId(), now);
```

Comment:
This is where the automated test is failing. The flow is to create a DAR, vote on it, etc. The failure here is that there is no saved collection with the id that's saved on the DAR. 

Flow:
* ElectionService updateElectionById ->
  * ElectionService updateSortDate ->
    * DataAccessRequestService updateByReferenceIdVersion2 ->  
      * DarCollectionDAO updateDarCollection

Reply:
This code is triggered when saving a draft DAR as well. Since we have DARs with a collection id of `0` from a different bug, the update fails with an NPE

Reply:
So the mapper bug ultimately causes the failure because it assigns a 0 to `collection_id`, which gets past all of the null checks in the code, allowing any query on `collection_id = 0` to fire off (and ultimately fail)?

Reply:
It causes one of the failures that I've seen. This point of failure is being hit from two code paths. One is from saving draft DARs (and the mapper bug is the source). The second one is from updating an election which then tries to subsequently update the dar collection. I'm still trying to understand why that's failing. It's calling this line with a collection id when it shouldn't have one to begin with ... because we're not saving DARs with collection ids yet.